lib/ansible/modules/cloud/vmware/vmware_cluster_ha.py

```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2015, Joseph Callen <jcallen () csc.com>
+# Copyright: (c) 2018, Ansible Project
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = r'''
+---
+module: vmware_cluster_ha
+short_description: Manage HA on VMware vSphere clusters
+description:
+    - This module can be used to manage HA on VMware vSphere clusters.
+    - All values and VMware object names are case sensitive.
+version_added: '2.9'
+author:
+- Joseph Callen (@jcpowermac)
+- Abhijeet Kasurde (@Akasurde)
+requirements:
+    - Tested on ESXi 5.5 and 6.5.
+    - PyVmomi installed.
+options:
+    cluster_name:
+      description:
+      - The name of the cluster to be managed.
+      type: str
+      required: yes
+    datacenter:
+      description:
+      - The name of the datacenter.
+      type: str
+      required: yes
+      aliases: [ datacenter_name ]
+    enable_ha:
+      description:
+      - If set to C(yes) will enable HA when the cluster is created.
+      type: bool
+      default: 'no'
+    ha_host_monitoring:
+      description:
+      - Indicates whether HA restarts virtual machines after a host fails.
+      - If set to C(enabled), HA restarts virtual machines after a host fails.
+      - If set to C(disabled), HA does not restart virtual machines after a host fails.
+      - If C(enable_ha) is set to C(no), then this value is ignored.
+      type: str
+      choices: [ 'enabled', 'disabled' ]
+      default: 'enabled'
+      version_added: 2.8
+    ha_vm_monitoring:
+      description:
+      - Indicates the state of virtual machine health monitoring service.
+      - If set to C(vmAndAppMonitoring), HA response to both virtual machine and application heartbeat failure.
+      - If set to C(vmMonitoringDisabled), virtual machine health monitoring is disabled.
+      - If set to C(vmMonitoringOnly), HA response to virtual machine heartbeat failure.
+      - If C(enable_ha) is set to C(no), then this value is ignored.
+      type: str
+      choices: ['vmAndAppMonitoring', 'vmMonitoringOnly', 'vmMonitoringDisabled']
+      default: 'vmMonitoringDisabled'
+      version_added: 2.8
```

Comment:
Should we mark this as added in 2.8? This functionality in this module didn't exist in 2.8.

Reply:
Changed to 2.9.